### PR TITLE
Add initial package-config generation 

### PR DESCRIPTION
**Pull request by sladyn98**
This pull request adds package-config generation. We would like to pass the form data that we get from the front end in the form of a JSON Object rather than put in tons of arguments therefore the argument that `generatePackageConfig` takes is a JSONObject and from that object we would construct the yaml file.

- [x]  Added tests

- [x] Run the build locally

### CODE REVIEW

**Comment by sladyn98** [comment]
Just some feedback if I am in the right direction would be great

File: pom.xml [hunk 1]
```diff
@@ -40,6 +40,12 @@
 				</exclusion>
 			</exclusions>
 		</dependency>
+		<dependency>
+			<groupId>com.vaadin.external.google</groupId>
+			<artifactId>android-json</artifactId>
```
Comment (kwhetstone) [inline pom.xml:45]: I've never seen this package before; is it specific to android development?  The first json package I thought of was the https://mvnrepository.com/artifact/org.json/json/20200518 package

File: src/main/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/Util/Util.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.org.jenkins.Custom.Jenkins.Distribution.Service.Util;
```
Comment (kwhetstone) [inline src/main/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/Util/Util.java:1]: Like the code quality check says!

File: src/test/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+import static com.org.jenkins.Custom.Jenkins.Distribution.Service.generators.PackageConfigGenerator.generatePackageConfig;
+
+public class PackageConfigGeneratorTest {
+    Util util = new Util();
+
+    @Test
+    public void testPackageConfigGeneration() throws Exception {
```
Comment (kwhetstone) [inline src/test/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGeneratorTest.java:15]: You probably want to try/catch and `assertFail` when you catch an exception during testing.  Definitely add a check for the expected package config.

File: src/main/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGenerator.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class PackageConfigGenerator {
+
+
+    public static String generatePackageConfig(JSONObject formData) throws JSONException {
+
+        //Get the bundle object
+        JSONObject bundle = formData.getJSONObject("bundle");
```
Comment (kwhetstone) [inline src/main/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGenerator.java:11]: I can imagine that you'll want to pull this section out into a method or a class that focuses on simply generating the information behind the war.  It doesn't hurt to pull this into a method so it's easier to test.

File: pom.xml (hunk 1 above)
Comment (sladyn98) [reply pom.xml:45]: Yeah I guess that should be the correct one, I should be more careful when importing :)

Comment (sladyn98) [reply pom.xml:45]: Changed :+1:

File: src/test/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGeneratorTest.java (hunk 3 above)
Comment (sladyn98) [reply src/test/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGeneratorTest.java:15]: Yeah just working on that.

File: src/main/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGenerator.java (hunk 4 above)
Comment (sladyn98) [reply src/main/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGenerator.java:22]: Done :+1:

**Comment by sladyn98** [comment]
I read in one of the blogs that it is not a good idea to test private methods, so IMO we should just write more tests via the public method.

File: src/main/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/Util/Util.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
```
Comment (oleg-nenashev) [inline src/main/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/Util/Util.java:1]: Codacy Issue found: [Package name contains upper case characters](https://app.codacy.com/gh/jenkinsci/custom-distribution-service/pullRequest?prid=5620628)

File: src/main/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGenerator.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
+package com.org.jenkins.Custom.Jenkins.Distribution.Service.generators;
```
Comment (oleg-nenashev) [inline src/main/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGenerator.java:1]: Codacy Issue found: [Package name contains upper case characters](https://app.codacy.com/gh/jenkinsci/custom-distribution-service/pullRequest?prid=5620628)

File: src/test/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGeneratorTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
+package com.org.jenkins.Custom.Jenkins.Distribution.Service.generators;
```
Comment (oleg-nenashev) [inline src/test/java/com/org/jenkins/Custom/Jenkins/Distribution/Service/generators/PackageConfigGeneratorTest.java:1]: Codacy Issue found: [Package name contains upper case characters](https://app.codacy.com/gh/jenkinsci/custom-distribution-service/pullRequest?prid=5620628)

File: src/main/java/com/org/jenkins/custom/jenkins/distribution/service/Util/Util.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package com.org.jenkins.custom.jenkins.distribution.service.Util;
```
Comment (oleg-nenashev) [inline src/main/java/com/org/jenkins/custom/jenkins/distribution/service/Util/Util.java:1]: Codacy Issue found: [Package name contains upper case characters](https://app.codacy.com/gh/jenkinsci/custom-distribution-service/pullRequest?prid=5620628)

**Comment by sladyn98** [comment]
@kwhetstone This pull request looks good to go.

File: pom.xml [hunk 9]
```diff
@@ -40,6 +40,30 @@
...
+			<groupId>com.fasterxml.jackson.dataformat</groupId>
+			<artifactId>jackson-dataformat-yaml</artifactId>
+			<version>2.9.9</version>
+		</dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-api</artifactId>
+      <version>5.6.2</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (LinuxSuRen) [inline pom.xml:59]: Please format the indent here.

File: src/test/resources/packagerConfig/emptyDockerTag.json [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  "plugins" : [{
+    "AnchorChain" : {
+      "version" : "4.12"
+    },
+    "gitlab-plugin" : {
+      "version" : "4.12"
+    }
+  }],
+  "casc" : true
+}
```
Comment (LinuxSuRen) [inline src/test/resources/packagerConfig/emptyDockerTag.json:27]: I would suggest that give it a new empty line of every files.

File: src/test/resources/packagerConfig/simpleConfig.json [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+  "plugins" : [{
+    "AnchorChain" : {
+      "version" : "4.12"
+    },
+    "gitlab-plugin" : {
+      "version" : "4.12"
+    }
+  }],
+  "casc" : true
+}
```
Comment (LinuxSuRen) [inline src/test/resources/packagerConfig/simpleConfig.json:28]: I would suggest that give it a new empty line of every files.

File: src/test/resources/packagerConfig/emptyCascSection.json [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  "plugins" : [{
+    "AnchorChain" : {
+      "version" : "4.12"
+    },
+    "gitlab-plugin" : {
+      "version" : "4.12"
+    }
+  }],
+  "casc" : false
+}
```
Comment (LinuxSuRen) [inline src/test/resources/packagerConfig/emptyCascSection.json:27]: I would suggest that give it a new empty line of every files.

File: pom.xml [hunk 13]
```diff
@@ -11,7 +11,7 @@
 	<groupId>io.jenkins.services.distribution-service</groupId>
 	<artifactId>custom-distribution-service</artifactId>
 	<version>0.0.1</version>
-	<name>Custom Jenkins Distribution Service</name>
+	<name>custom jenkins distribution service</name>
```
Comment (LinuxSuRen) [inline pom.xml:14]: I'm not sure why you do the changes here.

Comment (sladyn98) [reply pom.xml:14]: Yes I am changing this back, just a minor correcting while updating package names

**Comment by sladyn98** [comment]
> There's one more thing that might not be perfect. The `updateCenter.json` file is a little bit big, it will increase the git repository. Importantly `updateCenter.json` can be stale after a few days. It always being updated.

This PR deals with adding the downloader https://github.com/jenkinsci/custom-distribution-service/pull/31. I guess for running the tests we should just put a short version of the update center

File: pom.xml (hunk 9 above)
Comment (sladyn98) [reply pom.xml:59]: It is already indented.Is there any different way to indent ?

Comment (kwhetstone) [reply pom.xml:58]: The indent for this `dependency` is different than the others.  I think your editor might be able to auto-fix this.

**Comment by kwhetstone** [comment]
Looks like it's not iding test files because the test's aren't running: script returned exit code 143  I think that can addressed in a separate PR

File: pom.xml (hunk 9 above)
Comment (sladyn98) [reply pom.xml:58]: Oh I just noticed it shows up fine in my IDE.I will have a look

**Pull request merged**